src/node_contextify.cc

```diff
@@ -1099,12 +1099,23 @@ void ContextifyContext::CompileFunction(
           env->cached_data_string(),
           buf.ToLocalChecked()).IsNothing()) return;
     }
+
+    // Report if cache was produced.
```

Comment:
I looked into the implementation a bit, and I wonder if we can just wrap `ScriptCompiler::CreateCodeCacheForFunction` in another binding that takes a function as argument? That way we can just manipulate the return values in the JS land, and also enable the code cache generator to use the exact function compiled to generate the cache if we simply return `fn` in `NativeModule.prototype.compile` - also, we don't actually need that argument handling complexity in the `compileFunction` C++ binding, `vm.compileFunction` can handle `produce_cached_data` in JS land and `NativeModule.prototype.compile` doesn't even need it at all.

Reply:
Also I just noticed that the docs of `vm.compileFunction` did not mention its return value. Is that intentional?

Reply:
(and the returned values are not tested either?)